pkg/types/config/context.go

```diff
@@ -521,6 +523,12 @@ func NewUserContext(scaledContext *ScaledContext, config rest.Config, clusterNam
 	}
 	context.RBACw = rbacw.Rbac().V1()
 
+	ctlg, err := catalog.NewFactoryFromConfigWithOptions(&context.RESTConfig, &catalog.FactoryOptions{SharedControllerFactory: controllerFactory})
```

Comment:
Just to clarify my own understanding:

If I understand the purpose of this PR correctly, the issue is that an app created by apps and marketplace v2 (an object that is stored on the downstream clusters) can potentially conflict with a user-deployed app created by apps and marketplace v1 (which is validated via norman and stored in the Rancher cluster?).

So we are modifying the user context to add in the ability to query apps & marketplace v2 catalogs, which then allows us to add ClusterManager to the norman wrapper that is used for apps & marketplace v1 apps, so that on seeing a v1 app get created in the local cluster, we can check if a v2 app is already installed in the **downstream** cluster first before updating?

e.g. if I have Monitoring V2 installed under the release name `rancher-monitoring` in the downstream cluster, this prevents Monitoring V1 from being installed since `rancher-monitoring` already exists in the downstream cluster?

Let me know if I'm understanding this incorrectly; if not, the overall logic here looks good to me.

Reply:
That sounds right.